### ISSUE TITLE
Replies not sent back to client

### ISSUE THREAD

**Issue by Findorgri** [nodogsplash/nodogsplash#572]
I have posted this issue here: https://unix.stackexchange.com/questions/593705/nodogsplash-stopped-providing-internet-for-authenticated-clients
I guess I should have posted here from the start. I'm beginning to think that this may have something to do with the iptables rules added by nodogsplash.
I'm posting them here in case they are needed:
```
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-N ndsNET
-N ndsRTR
-N ndsAUT
-N ndsTRU
-N ndsTRT
-A INPUT -i wlan0 -j ndsRTR
-A FORWARD -i wlan0 -j ndsNET
-A ndsNET -m mark --mark 0x10000/0x30000 -j DROP
-A ndsNET -m conntrack --ctstate INVALID -j DROP
-A ndsNET -p tcp -m tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
-A ndsNET -m mark --mark 0x20000/0x30000 -j ACCEPT
-A ndsNET -m mark --mark 0x30000/0x30000 -j ndsAUT
-A ndsNET -j REJECT --reject-with icmp-port-unreachable
-A ndsNET -j REJECT --reject-with icmp-port-unreachable
-A ndsRTR -m mark --mark 0x10000/0x30000 -j DROP
-A ndsRTR -m conntrack --ctstate INVALID -j DROP
-A ndsRTR -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A ndsRTR -p tcp -m tcp ! --tcp-option 2 --tcp-flags SYN SYN -j DROP
-A ndsRTR -p tcp -m tcp --dport 2050 -j ACCEPT
-A ndsRTR -m mark --mark 0x20000/0x30000 -j ACCEPT
-A ndsRTR -p udp -m udp --dport 53 -j ACCEPT
-A ndsRTR -p tcp -m tcp --dport 53 -j ACCEPT
-A ndsRTR -p udp -m udp --dport 67 -j ACCEPT
-A ndsRTR -p tcp -m tcp --dport 22 -j ACCEPT
-A ndsRTR -p tcp -m tcp --dport 80 -j ACCEPT
-A ndsRTR -p tcp -m tcp --dport 443 -j ACCEPT
-A ndsRTR -j REJECT --reject-with icmp-port-unreachable
-A ndsAUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A ndsAUT -j ACCEPT
-A ndsAUT -j REJECT --reject-with icmp-port-unreachable
```
I'd be happy to provide any other details needed.

**Comment by bluewavenet** [comment]
@Findorgri 
Please supply:
 * hardware details
 * operating system in use
 * version of NoDogsplash
 * your Nodogsplash config file
 * confirmation you are running on a device configured as an IPv4 router with NAT, DHCP and DNS, with the lan interface acting as the default ip gateway

Note also that you should not make any custom iptables or firewall changes AFTER NoDogSplash has started. If you do, it will most likely fail.

**Comment by Findorgri** [comment]
I'm using a USB wireless card TP-Link WN722N V1
The operating system is a Kali Linux 2019.4 virtual machine
Nodogsplash version is 3.3.3-beta
In the .conf file I have changed only GatewayInterface, GatewayAddress and also added a rule in users-to-router `FirewallRule allow tcp port 4444`, but I have also tried it without the last one. I'm attaching the file.
As I mentioned in the link I provided, I'm using hostapd and dnsmasq to create a wireless router. The lan interface is the default gateway for accessing the Internet.
Everything worked fine at first setup of nodogsplash. The clients had Internet access after authentication. Now it seems that the virtual machine is forwarding the client's request to the Internet, but not forwarding the replies to the clients.
I'm not making changes in iptables after nodogsplash has started.

**Comment by bluewavenet** [comment]
@Findorgri 

> I'm using a USB wireless card TP-Link WN722N V1
> The operating system is a Kali Linux 2019.4 virtual machine

I cannot really comment on this other than NDS must be running on a device configured as an IPv4 router with **_NAT,_** DHCP and DNS, with the lan interface acting as the default ip gateway.

In your link you state you have to add a NAT rule to give access to the Internet - indeed - this is a requirement of the "router" that NDS is to be installed on.

If, as you say in your link, "after a few days" it stopped working, then I suggest your virtual machine is no longer running NAT.

Note also that NoDogSplash 3.3.3beta is a work in progress after the split to OpenNDS and not a release, so has most likely not been tested much, if at all. The most up to date release is openNDS v5.0.1 at the time of writing.

https://github.com/openNDS/openNDS/releases/tag/v5.0.1

**Comment by Findorgri** [comment]
I thought that by enabling NAT with this rule I mentioned, I was giving access to the Internet for every client. But iptables rules added by NDS block them until they are authenticated. It "stopped working" because when I first tried it I had NAT setup with this rule, but I failed to note it apparently. This was it, I had to have this iptables rule in place for it to work.
Thank you for your time.
On the matter of the version, when I do git pull, I get the 3.3.3-beta version. Should I download the one from the link you provided?

**Comment by bluewavenet** [comment]
@Findorgri 
After the split, NoDogSplash is to become a minimal functionality, minimal footprint captive portal for IoT devices with very limited resources, while openNDS is already a full implementation of a Captive Portal engine with api for development of sophisticated CP solutions (still with a relatively small footprint).
openNDS is released and in active development with version 5.1.0 release imminent, while the new NoDogSplash "minimalistic" version is not yet released.
Yes, download from the link I provided, or alternatively run:
`git clone -b 5.0.1 https://github.com/opennds/opennds.git`
